controllers/note.py

```diff
@@ -0,0 +1,107 @@
+import math
+
+from magicbot import StateMachine, state, timed_state, will_reset_to
+from wpimath.geometry import Translation2d
+
+from components.chassis import ChassisComponent
+from components.injector import InjectorComponent
+from components.intake import IntakeComponent
+from components.shooter import ShooterComponent
+from utilities.game import get_goal_speaker_position
+
+
+class NoteManager(StateMachine):
+
+    shooter_component: ShooterComponent
+
+    chassis: ChassisComponent
+    injector_component: InjectorComponent
+    intake: IntakeComponent
+
+    shot_desired = will_reset_to(False)
+    intake_desired = will_reset_to(False)
+    intake_cancel_desired = will_reset_to(False)
+
+    def try_intake(self):
+        self.intake_desired = True
+
+    def cancel_intake(self):
+        self.intake_cancel_desired = True
+
+    def try_shoot(self):
+        self.shot_desired = True
+
+    def has_note(self):
+        return self.injector_component.has_note()
+
+    @property
+    def translation_to_goal(self) -> Translation2d:
+        return (
+            get_goal_speaker_position().toTranslation2d()
+            - self.chassis.get_pose().translation()
+        )
+
+    @state(must_finish=True)
+    def idling(self):
```

Comment:
I think this should probably be the `default_state`. The driver should probably hold a button to intake, so if we make the two intake states not `must_finish`, then not engaging the state machine would mean it would naturally revert to the default state.

Reply:
The drivers have requested a two button toggle rather than a button holding system. The system also has two "default states" as we want to return to either idling or holding note depending on how we exit the other states.